fms/modules/positions.py

```diff
@@ -135,105 +238,34 @@ def __init__(
         self.scaling = copy.deepcopy(scaling)
         if "rope_type" not in self.scaling:
             self.scaling["rope_type"] = "regular"
+        self.rope_scaling = _rope_scale_mapping[self.scaling["rope_type"]](self)
```

Comment:
I think the self.scaling should move into the RopeScaling subclass, as well ass the max_seq_len. If you also move dim and ratio there, and change needs_scaling to a function that simply computed the "scaled_max_seq_length" and do the needs scaling check in the main class, this should remove the circular dependency